modules/basis/src/lib/parse-basis.js

```diff
@@ -1,8 +1,38 @@
+import {BasisFormat} from '../basis-loader';
 import {loadBasisModule} from './basis-module-loader';
 
 async function parse(data, options) {
   const {BasisFile} = await loadBasisModule(options);
-  return new BasisFile(new Uint8Array(data));
+  const basisFile = new BasisFile(new Uint8Array(data));
+
+  try {
+    const width = basisFile.getImageWidth(0, 0);
+    const height = basisFile.getImageHeight(0, 0);
+    const images = basisFile.getNumImages();
+    const levels = basisFile.getNumLevels(0);
+    const hasAlpha = basisFile.getHasAlpha();
+
+    let format = (options && options.format) || BasisFormat.cTFRGB565;
+    if (typeof format === 'object') {
+      format = hasAlpha ? format.alpha : format.noAlpha;
+    }
+
+    if (!basisFile.startTranscoding()) {
+      return null;
+    }
+
+    const decodedSize = basisFile.getImageTranscodedSizeInBytes(0, 0, format);
+    const decodedData = new Uint8Array(decodedSize);
+
+    if (!basisFile.transcodeImage(decodedData, 0, 0, format, 0, 0)) {
+      return null;
+    }
+
+    return {width, height, images, levels, hasAlpha, format, decodedSize, decodedData};
```

Comment:
You'll probably want to parse out the separate mip levels as they have to be loaded separately.  Here's how I do it in the PicoGL demos: https://github.com/tsherif/picogl.js/blob/master/examples/utils/utils.js#L680-L704

Key thing is that the size of a level depends on the format.

Reply:
BASIS decodes each level separately in decodedData.
So this will just only have level 0.

Reply:
I need a .basis with multiple mip levels to test. Do you have any?

Reply:
OK, rewrote that function.